script.js

```diff
@@ -1,4 +1,117 @@
+// Define current player
+var currentPlayer = 1;
+
+// Function for output
+var myOutputValue = "";
+
+// Define variables to keep track of player's number
+var playerNumber = [];
+var playerOneNum = playerNumber[0];
+var playerTwoNum = playerNumber[1];
+
+// Function to track game mode
+var modeRollDice = "Roll dice mode";
+var modeChoseOrder = "Chose order of numbers";
+var modeCurrent = modeRollDice;
+
+// Function to keep track of dice rolled
+var diceNumberRolled = [];
+var diceRollA = "";
+var diceRollB = "";
+
+// Combination of numbers
+// var combineNumbers = "";
+var combineNumbersAB = Number(diceRollA) + Number(diceRollB);
+var combineNumbersBA = Number(diceRollB) + Number(diceRollA);
+
+// Function to get random dice number
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // Select game mode
+  var gameCounter = 0;
+  if (modeCurrent == modeRollDice) {
+    myOutputValue = rollDiceTwice(gameCounter);
+    modeCurrent = modeChoseOrder;
+  }
+  // Choose order of numbers
+  else if (modeCurrent == modeChoseOrder) {
```

Comment:
The logic inside this game state can be modularised into a function, similar to what you are already doing with [execution of the `rollDiceTwice` function, where you change the myOutputValue to the string returned from the function](https://github.com/rocketacademy/basics-beat-that/blob/e3e941a6585a42407eb2a4218d945119427bf058/script.js#L102-L117). As you can see [here](https://github.com/rocketacademy/basics-beat-that/blob/e3e941a6585a42407eb2a4218d945119427bf058/script.js#L43-L72), the end result of the function (or its `return`) can simply be a string that is `return`ed back to the grey output box.